test/functionalTest/testHarness.sh

```diff
@@ -638,8 +638,25 @@ function runTest()
 
   if [ "$exitCode" != "0" ]
   then
-    exitFunction 11 "SHELL-INIT exited with code $exitCode" $path "($path)" "" DIE
-    return
+    # 3.2 Run the SHELL-INIT part AGAIN
```

Comment:
I would add some sentence about why re-running SHELL-INIT is need (something like "this step may be needed due to glitches that we suspect are due to one test starts before the former has ended).